src/devices/machine/atmel_arm_aic.h

```diff
@@ -8,17 +8,14 @@
 
 DECLARE_DEVICE_TYPE(ARM_AIC, arm_aic_device)
 
-#define MCFG_ARM_AIC_ADD(_tag) \
-	MCFG_DEVICE_ADD(_tag, ARM_AIC, 0)
-
 #define MCFG_IRQ_LINE_CB(_devcb) \
 	downcast<arm_aic_device &>(*device).set_line_callback(DEVCB_##_devcb);
 
 class arm_aic_device : public device_t
 {
 public:
 	// construction/destruction
-	arm_aic_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock) :
+	arm_aic_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock = 0) :
```

Comment:
Does this device really not have a clock source?  If it logically has a clock, it shouldn't be defaulted to zero just for convenience.  I'd expect a peripheral like this probably uses the bus clock or core clock of the CPU it's associated with.